userspace/libsinsp/test/string_visitor.ut.cpp

```diff
@@ -0,0 +1,261 @@
+/*
```

Comment:
This looks great! I think my only suggestion would be to turn merge all the `TEST_F` for the unidirectional/bidirectional case. In all cases, you could have only one  `TEST_F` that calls `unidirectional()` first and `bidirectional()` after. This is just a minor comment to make the code less verbose, feel free to ignore it if you like it this way!

Reply:
It's a bit of a preference, but I prefer having each TEST/TEST_F only test a single test case, as it's easier to pinpoint the failure when a test fails.